Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.h

```diff
@@ -552,6 +552,8 @@ class AXIsolatedObject final : public AXCoreObject {
     Vector<AXID> m_childrenIDs;
     Vector<RefPtr<AXCoreObject>> m_children;
     AXPropertyMap m_propertyMap;
+    // A bitfield for some of the most common bool properties
+    uint32_t m_propertyFlags;
```

Comment:
This is a clever and useful optimization. I would suggest for maintainability and clarity sake to consider the following implementation approach:
1. Leave the AXPropertyName enum as is.
2. Add a new enum with powers of 2 called something like CommonBoolPropertyBit. This can be a private enum in AXIsolatedObject.
3. A simple helper function to return a CommonBoolPropertyBit given a PropertyName or 0 if not a common bool. This would be essentially a switch statement.
4. Make m_propertyFlags and WTF::OptionSet<CommonBoolPropertyBit>.
5. Use WTF::OptionSet methods for manipulation of the bits.
I think this would make it more clear and robust down the road, and still be a good optimization.

Reply:
Yeah, my original version was like that. After Tyler's feedback I decided to try to optimize it more but maybe it's less maintainable.

I put it back to the original version that uses OptionSet and switch statements. Let me know what you think.